Odysee/Utils/Multistream.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  Multistream.swift
+//  Odysee
+//
+//  Created by Adlai on 5/21/21.
+//
+
+import Foundation
+
+fileprivate let kBufferSize = 1024 * 1024
+
+class Multistream: InputStream, StreamDelegate {
+    private let streams: [InputStream]
+    private var nextInputIndex = 0
+    private var currentInputIndex: Int?
+    private var input: InputStream
+    private var output: OutputStream
+    private var buffer = Data(count: kBufferSize)
+    private var bufferLength = 0
+    var error: Error?
+
+    // MARK: - Pass-through to underlying bound stream
+
+    override var streamStatus: Stream.Status {
+        return input.streamStatus
+    }
+    override func open() {
+        input.open()
+        output.open()
+    }
+    override func close() {
+        input.close()
+    }
+    override func setProperty(_ property: Any?, forKey key: Stream.PropertyKey) -> Bool {
+        input.setProperty(property, forKey: key)
+    }
+    override func property(forKey key: Stream.PropertyKey) -> Any? {
+        return input.property(forKey: key)
+    }
+    override func schedule(in aRunLoop: RunLoop, forMode mode: RunLoop.Mode) {
+        input.schedule(in: aRunLoop, forMode: mode)
+        output.schedule(in: aRunLoop, forMode: mode)
+    }
+    override var delegate: StreamDelegate? {
+        get {
+            return input.delegate
+        }
+        set {
+            input.delegate = newValue
+        }
+    }
+    override var hasBytesAvailable: Bool {
+        return input.hasBytesAvailable
+    }
+    override func read(_ buffer: UnsafeMutablePointer<UInt8>, maxLength len: Int) -> Int {
+        assert(input.streamStatus == .open)
+        return input.read(buffer, maxLength: len)
+    }
+    override func getBuffer(_ buffer: UnsafeMutablePointer<UnsafeMutablePointer<UInt8>?>,
+                            length len: UnsafeMutablePointer<Int>) -> Bool {
+        return input.getBuffer(buffer, length: len)
+    }
+
+    // Streams should be open before this.
+    // Don't mess with the streams otherwise.
+    init(streams: [InputStream]) {
+        self.streams = streams
+        assert(!streams.contains { $0.streamStatus != .open})
+        var read: Unmanaged<CFReadStream>?
+        var write: Unmanaged<CFWriteStream>?
+        CFStreamCreateBoundPair(nil, &read, &write, kBufferSize)
+        input = read!.takeRetainedValue()
+        output = write!.takeRetainedValue()
+        super.init()
```

Comment:
The build fails on this line with an error because `InputStream` does not have an empty initialiser. I changed this to `super.init(data: Data())` which works. I'll go ahead and merge if this is fine.

Reply:
Weird. What version of Xcode are you using? I’m on 12.5 and it doesnt mind. I’ll push the change anyway so we’re unblocked.

Reply:
12.4. I guess 12.5 isn't available for macOS Catalina.